index.js

```diff
@@ -486,4 +483,31 @@ FeatureService.prototype._catchErrors = function (task, e, url, cb) {
   }.bind(this), task.retry * 1000)
 }
 
+/* Forms a standardized error object for reporting back up the chain
+ * @param {object} error - http request error
+ * @param {string} message - human readable message about ther
+ * @param {object} json - json returned from server request
+ * @returns {object} - error object containing diagnostic information
+ */
+FeatureService.prototype._buildError = function (url, message, error) {
+  if (!error) error = {}
+  error.message = message
+  error.request = url
+  error.response = error.message
+  error.type = error.type || 'Unknown failure'
+  error.code = error.code || 500
+
+  return error
+}
```

Comment:
Couple of things here:
- I could use error if that will allow me to get the properties out of it
- That is a mistake, it's supposed to be the error message from the GIS server and then a message from us explaining what happened
- I can make types better, or I can get rid of it.

Please do take a crack at it and show me an example